### ISSUE TITLE
onManaChange for spells

### ISSUE THREAD

**Issue by Zbizu** [otland/forgottenserver#1211]
onManaChange is not executed when I use spells.
my script is registered correctly, because it works when a monster attacks with manadrain

**Comment by dalkon** [comment]
onManaChange does not execute whenever the origin is ORIGIN_NONE, and I believe that it is that when you use spells.

**Comment by Zbizu** [comment]
so it works as intended, right?

**Comment by dalkon** [comment]
Yup.

**Comment by dalkon** [comment]
Actually both onHealthChange and onManaChange could be extended to execute for more things, like when you use a spell as you said.

**Comment by Tofame** [comment]
I think its incorrect to call onManaChange() only in `combatChangeMana`
It should be called in changeMana() methods as its the most logical && appropiate thing.
More origins could be added so people would know that something is actually from a spell, combat or OTHER sources.

**Comment by Codinablack** [comment]
This can be closed. While onManaChange sounds like it's logical to be executed when mana is spent on spells, it's not practical. Having played around with this specifically, I can tell you that the exact right location for an event like onManaChange to occur when spending spells, is the exact location that onGainSkillTries is called... furthermore, because onManaChange might sound like an all around event for handling mana, that's not what it actually is, it's a combat event, which is why you must pass it combat damage as a parameter on the backend, and return the damage numbers on the frontend... so when you would call this event anyways, you would have to create an combat damage object to pass along a mana change..

TLDR

Use onGainSkillTries() for hooking into mana changes caused from the player spending the mana on a spell.
